plugins/tracers/gstproctime.c

```diff
@@ -66,24 +66,37 @@ static void
 do_push_buffer_pre (GstTracer * self, guint64 ts, GstPad * pad)
 {
   GstProcTimeTracer *proc_time_tracer;
+  GstSharkTracer *shark_tracer;
   GstProcTime *proc_time;
 
   GstPad *pad_peer;
+  GstElement *element;
+  gchar *element_name;
   gchar *name;
   GstClockTime time;
   gchar *time_string;
   gboolean should_log;
+  gboolean should_calculate;
 
   proc_time_tracer = GST_PROC_TIME_TRACER (self);
+  shark_tracer = GST_SHARK_TRACER (proc_time_tracer);
   proc_time = proc_time_tracer->proc_time;
   name = GST_OBJECT_NAME (GST_OBJECT_PARENT (pad));
 
+  element = gst_pad_get_parent_element (pad);
```

Comment:
The variable `name` already contains the value you are computing for `element_name`